### ISSUE TITLE
Make http://citationstyles.org/ and subdomains HTTPS

### ISSUE THREAD

**Issue by adam3smith** [citation-style-language/citation-style-language.github.io#37]
https://blog.cloudflare.com/secure-and-fast-github-pages-with-cloudflare/

Who has control over the registration (which is needed for this)? Is that still at CHNM?

**Comment by rmzelle** [comment]
There are indications that GitHub is (finally!) starting to slowly roll out HTTPS support for custom domains, so I would just wait and not bother with the Cloudflare hack. See https://github.com/isaacs/github/issues/156#issuecomment-366543897 for recent progress on this issue.

For http://docs.citationstyles.org/ we're dependent on hosting by https://readthedocs.org/, so that's a separate issue, see https://github.com/rtfd/readthedocs.org/issues/2652#issuecomment-281140560.

(P.S. https://discourse.citationstyles.org/ is also already HTTPS)

**Comment by rmzelle** [comment]
Some good news already: I was able to enable "Enforce HTTPS" for https://formatter.citationstyles.org/ and https://validator.citationstyles.org/, which are now served over HTTPS. Per the issue linked above, it looks like GitHub is a little further along with third-level hostnames like https://formatter.citationstyles.org/ than with second-level hostnames like http://citationstyles.org/ (https://github.com/isaacs/github/issues/156#issuecomment-369637020). Can't enable HTTPS for http://citationstyles.org/ yet.

**Comment by jayvdb** [comment]
You may want to try https://netlify.com which provides this feature already, and more.

**Comment by rmzelle** [comment]
> You may want to try https://netlify.com which provides this feature already, and more.

I rather not, especially since GitHub seems to be finally rolling this out, probably driven by Google's decision to mark all HTTP sites as non-secure by July 2018: https://security.googleblog.com/2018/02/a-secure-web-is-here-to-stay.html. I assume we'll get native HTTPS support for GitHub Pages if we just wait another month or two.

**Comment by rmzelle** [comment]
Earlier today GitHub enabled HTTPS for all custom domains, so we've just activated it for https://citationstyles.org/. Dan Stillman from Zotero kindly updated the A record (domain is still at CHNM), and I had to fix a small mixed content issue with our theme (solved by specifying the HTTPS domain in _config.yml per https://github.com/mmistakes/minimal-mistakes/issues/1318#issuecomment-339642335), but everything seems to work now. I enabled "Enforce HTTPS".

So the only remaining HTTP part is http://docs.citationstyles.org/, which needs a change at Read the Docs (as I wrote above at https://github.com/citation-style-language/citation-style-language.github.io/issues/37#issuecomment-372047627).
